deepmd/utils/data_system.py

```diff
@@ -103,6 +100,7 @@ def __init__(
         del rcut
         self.system_dirs = systems
         self.nsystems = len(self.system_dirs)
+        assert self.nsystems > 0, "No systems provided"
```

Comment:
`assert` should not be used; see https://docs.astral.sh/ruff/rules/assert/

Reply:
Make sense; I was not aware of it before.